src/Rings/mpoly-ideals.jl

```diff
@@ -933,7 +935,11 @@ false
 ```
 """
 function ideal_membership(f::T, I::MPolyIdeal{T}; ordering::MonomialOrdering = default_ordering(base_ring(I))) where T
-  GI = standard_basis(I, ordering=ordering, complete_reduction=false)
+  if isempty(I.gb)
+    GI = standard_basis(I, ordering=ordering, complete_reduction=false)
+  else
+    GI = first(I.gb)[2]
```

Comment:
Ah but this now ignores `ordering`? Is this temporary?

Moving forward, we discussed a bit last week, and it seems that a lot of the problems could be addressed if only `MonomialOrdering` cached its hash value and perhaps a few other things, *and* if the the default ordering was actually stored in the ring, so that calling `default_ordering(base_ring(I))` twice returns an identical ordering. Does that sound sensible?

Reply:
@fingolfin has a point here -- though not the one he intended to make:
If we are only considering global orderings, we are free to take any ordering and a standard basis w.r.t. it to perform the test. So if a standard basis has already been computed, we should not recompute. In this case, @ederc is right.
However: As we also allow **local** and **mixed** orderings, we need to make sure that the given ordering and the ordering of the stored standard basis have 1>x_i for exactly the same variables x_i. This might require checking more than just the first standard basis in the list for a suitable ordering to appear (think of deliberately computing ideal membership in the polynomial ring and then in its localization at the origin to spot a difference!)

Reply:
It sounds like these orderings model different rings and different mathematics. 
Mixing them in the same parent seems unnatural to me. Or at least the output should be an ideal over a the new appropriate parent.

Reply:
Yes, indeed. We are suffering from a design difference between Singular and Oscar here: 
- Singular controls global and local computations by means of the monomial ordering -- and it is (and needs to be) possible to specify local orderings for the Singular side also from Oscar (because we need this functionality). 
- Oscar deliberately makes a (mathematically well founded) distinction between Polynomial Rings and their localization, but relies on functionality of Singular for computations in the latter. 

The challenge here is to make sure that two currently unrelated choices do not lead to contradictions: 
- the expectations of the (programming) user, who might have deliberately specified a global or a local ordering to pass to the backend
- the returned result, which currently happens to rely on the (inherited and thus uncontrolled) choice of the ordering of the first precomputed standard basis.